src/database/database.ts

```diff
@@ -83,3 +106,56 @@ export class SupaBaseDatabase {
     }
   }
 }
+
+export const getCachedAnswer = async (question: string, author?: string) => {
+  question = question.toLowerCase();
+  author = author?.toLocaleLowerCase();
+  const answers = await SupaBaseDatabase.getInstance().getAnswerByQuestion(
+    question,
+    author
+  );
+
+  if (!answers || answers.length === 0) {
+    console.error("Error fetching answer: No answers found.");
+    return null;
+  }
+
+  // Use JavaScript .find() method to get first element where answer is not an empty string
+  const nonEmptyAnswer = answers.find((item) => item.answer.trim() !== "");
+
+  if (!nonEmptyAnswer) {
+    console.error("Error fetching answer: No non-empty answers found.");
+    return null;
+  }
+
+  // Return the nonEmptyAnswer directly as a string
+  return createReadableStream(nonEmptyAnswer.answer);
+};
+
+export const addDocumentToSupabase = async (payload: any) => {
+  await SupaBaseDatabase.getInstance().insertData(payload);
+};
+
+function createReadableStream(text: string) {
+  const encoder = new TextEncoder();
+  const readable = new ReadableStream({
+    start(controller) {
+      controller.enqueue(encoder.encode(text));
+      controller.close();
+    },
+  });
+  return readable;
+}
+
+export const addFeedback = async (feedback: FeedbackPayload, feedbackId: string) => {
+  const { status, error } = await SupaBaseDatabase.getInstance().addFeedback({
+    ...feedback,
```

Comment:
No need to spread the feedback object. Pass it directly.
```suggestion
    feedback,
```

Reply:
I get this from my linter when I change to your suggestion:

`Argument of type '{ feedback: FeedbackPayload; feedbackId: string; }' is not assignable to parameter of type 'FeedbackPayload'.
  Object literal may only specify known properties, but 'feedback' does not exist in type 'FeedbackPayload'. Did you mean to write 'feedbackId'?ts(2345)`

Reply:
Okay. its a type issue but this is fine too